e2e/gui/src/test/java/com/epam/pipeline/autotests/RestrictionsOnInstancePriceTypeTest.java

```diff
@@ -93,74 +110,562 @@ public void deletingEntities() {
+            tools()
+                    .perform(defaultRegistry, defaultGroup, testingTool, ToolTab::runWithCustomSettings)
+                    .sleep(1, SECONDS)
+                    .checkValueIsInDropDown(INSTANCE_TYPE, instanceFamilyName);
+        } finally {
+            logout();
+            loginAs(admin);
+            setMaskForUser(user.login, toolInstanceTypesMask, "");
+        }
+    }
+
+    @Test
+    @TestCase({"EPMCMBIBPC-2643"})
+    public void validationOfToolsInstanceTypesRestrictionsForUserGroup() {
+        try {
+            loginAs(admin);
+            setMaskForRole(testRole, toolInstanceTypesMask, format("%s.*", instanceFamilyName));
+            logout();
+            loginAs(user);
+            tools()
+                    .perform(defaultRegistry, defaultGroup, testingTool, ToolTab::runWithCustomSettings)
+                    .sleep(1, SECONDS)
+                    .checkValueIsInDropDown(INSTANCE_TYPE, instanceFamilyName);
+        } finally {
+            logout();
+            loginAs(admin);
+            setMaskForRole(testRole, toolInstanceTypesMask, "");
+        }
+    }
+
+    @Test
+    @TestCase({"EPMCMBIBPC-2642"})
+    public void validationOfToolsInstanceTypesRestrictionsOverInstanceManagement() {
+        try {
+            loginAs(admin);
+            tools()
+                    .performWithin(defaultRegistry, defaultGroup, testingTool, tool ->
+                            tool.showInstanceManagement(instanceManagement ->
+                                    instanceManagement
+                                            .addAllowedToolInstanceTypesMask(format("%s.*", instanceFamilyName))
+                                            .clickApply()
+                                            .sleep(2, SECONDS)));
+            tools()
+                    .perform(defaultRegistry, defaultGroup, testingTool, ToolTab::runWithCustomSettings)
+                    .expandTab(EXEC_ENVIRONMENT)
+                    .checkValueIsInDropDown(INSTANCE_TYPE, instanceFamilyName);
+            logout();
+            loginAs(user);
+            tools()
+                    .perform(defaultRegistry, defaultGroup, testingTool, ToolTab::runWithCustomSettings)
+                    .expandTab(EXEC_ENVIRONMENT)
+                    .checkValueIsInDropDown(INSTANCE_TYPE, instanceFamilyName);
+        } finally {
+            logout();
+            loginAs(admin);
+            tools()
+                    .performWithin(defaultRegistry, defaultGroup, testingTool, tool ->
+                            tool.showInstanceManagement(instanceManagement ->
+                                    instanceManagement
+                                            .addAllowedToolInstanceTypesMask("")
+                                            .clickApply()
+                                            .sleep(2, SECONDS)));
+        }
+    }
+
+    @Test(dependsOnMethods={"preparationForValidationOfInstanceTypesRestrictions"})
```

Comment:
Could you please add space before and after = ?

Reply:
done